src/libraries/icubmod/embObjLib/diagnosticInfoFormatter.cpp

```diff
@@ -157,4 +158,18 @@ bool EntityNameProvider::getAxisName(uint32_t entityId, std::string &axisName)
     }
     
     return (m_MC_ethRes->getEntityName(entityId, axisName));
+}
+
+bool EntityNameProvider::getEncoderTypeName(uint32_t jomoId, eOmc_position_t pos, std::string &encoderTypeName)
+{
+    if (m_MC_ethRes == nullptr)  
+    {
+        encoderTypeName = "N/A";
+        return false;
+    }
+
+    if(embObjMotionControl *emc = dynamic_cast<embObjMotionControl*>(m_MC_ethRes); emc != nullptr)
```

Comment:
To avoid circular dependency I would:
- add `getEncoderTypeName` in the [`IEthResource`](https://github.com/robotology/icub-main/blob/22c06ceffe66c21de446da841a1ffd432ad515d0/src/libraries/icubmod/embObjLib/IethResource.h) with a default implementation
- Implement (marking it `override`) `getEncorderTypeName` in `embObjMotionControl`
- Change this code in order to directly invoke `m_MC_ethRes->getEncoderTypeName` without the need of the dynamic_cast